packages/gestalt/src/contexts/I18nProvider.js

```diff
@@ -3,85 +3,80 @@ import { type Context, createContext, useContext } from 'react';
  */
 const I18nProvider = I18nContext.Provider;
 export default I18nProvider;
 
-type ValidComponent = $Keys<I18nContextType>;
+type ValidComponent = $Keys<typeof defaultTranslations>;
 
 export function useI18nContext<C: ValidComponent>(
   componentName: C,
 ): $ElementType<I18nContextType, C> {
+  // Without a I18nProvider set in the app, propsTranslations will contain the default values
+  // If a I18nProvider is set in the app, propsTranslations will contain the newly set context values and override the default ones
   const propsTranslations = useContext(I18nContext);
-  const componentTranslations = propsTranslations[componentName];
 
-  // No translation provided for the entire component even in initialContext
-  if (!componentTranslations) {
-    throw new Error(
-      `${componentName} translations not available — please add translations to I18nProvider`,
-    );
-  }
+  // Check if a I18nProvider is set in the app but the key-values for a particular component are missed
+  const providedTranslations = propsTranslations[componentName];
+
+  // Get the default key-values for a particular component
+  const defaultComponentTranslations = defaultTranslations[componentName];
+
+  // If all key-values for a particular component are missed in the I18nProvider, used the default ones for that component
+  if (!providedTranslations) return defaultComponentTranslations;
+
+  // Shallow cloning (copies first level, references second level) to support debugging
+  const componentTranslations = { ...providedTranslations };
 
   const translationPropNames = Object.keys(componentTranslations);
-  // Translations that are still nullish (less likely)
+
+  // Check if there's any values set in the I18nProvider that are nullish
   const nullTranslations = translationPropNames.filter(
     (propName) => typeof componentTranslations[propName] !== 'string',
   );
-  const expectedTranslations = Object.keys(initialContext[componentName]);
-  // Translations that weren't provided in the Provider (more likely)
+
+  const expectedTranslations = Object.keys(defaultComponentTranslations);
+
+  // Check if there's any particular key-values in a component that weren't provided in the I18nProvider but do exist as defaults
   const missingTranslations = expectedTranslations.filter(
     (item) => !translationPropNames.includes(item),
   );
+
   const missingAndNullTranslations = [...nullTranslations, ...missingTranslations];
 
+  // For those nullish and missing key-values, complete them with the default values
   if (missingAndNullTranslations.length > 0) {
-    const multipleMissing = missingAndNullTranslations.length > 1;
-    const isNotProd = process.env.NODE_ENV !== 'production';
-
-    if (isNotProd) {
-      // Ideally we would throw an actual error here (at least in dev), but this wasn't working with the mocks in Pinboard
-      // eslint-disable-next-line no-console
-      console.error(
```

Comment:
The problem with this is that we're not communicating to the user that translations are missing, and our default strings aren't translated. So we would be showing untranslated strings in the app with no indication that anything is wrong. I never wanted to provide default translations in the first place, but at least the previous logic would alert the user to the issue.